applications/models.py

```diff
@@ -232,7 +232,11 @@ def is_scored_by_user(self, user):
         Returns true if the given user has scored this application
         or false if they have not, or there is a zero score.
         """
-        return self.scores.filter(user=user, score__gt=0).exists()
+        try:
+            self._prefetched_objects_cache['scores']  # check if scores prefetched
+            return any(map(lambda s: s.user_id == user.id and s.score and s.score > 0, self.scores.all()))
+        except (AttributeError, KeyError):
```

Comment:
Thank you, it would be great to reduce the number of queries here.

I think `_prefetched_objects_cache` is an undocumented Django internal. To ensure we pick up on any changes that may happen in future versions of Django, and also to verify the logic here, this branch of the code should be covered by a test. Could you add one please?

The lambda would probably also be clearer as a named function.

Thanks! :sparkles: